libs/website/forms/src/lib/contact-form/contact-form.tsx

```diff
@@ -0,0 +1,143 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Button } from '../../../../ui-kit/src/lib/button/button';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { Modal } from '@haqq/ui-kit';
+import {
+  FormError,
+  FormFields,
+  FormState,
+  HookedFormInput,
+} from '../hooked-form-input/hooked-form-input';
+import { HookedFormTextarea } from '../hooked-form-textarea/hooked-form-textarea';
+import { SuccessMessageModal } from '@haqq/website/ui-kit';
+
+const schema = yup
+  .object({
+    name: yup.string().required('Name is required'),
+    email: yup
+      .string()
+      .email('Invalid email format')
+      .required('Email is required'),
+    message: yup.string(),
+  })
+  .required();
+
+function submitForm(form: FormFields): Promise<any> {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve({ status: 200 });
+    }, 2500);
+  });
+}
+
+export function ContactForm() {
+  const [contactFormState, setContactFormState] = useState<FormState>(
+    FormState.idle,
+  );
+  const [isMessageSent, setMessageSent] = useState<boolean>(false);
+
+  const { register, handleSubmit, formState } = useForm<FormFields>({
+    resolver: yupResolver(schema),
+  });
+
+  const onSubmit = useCallback(async (data: FormFields) => {
+    try {
+      setContactFormState(FormState.pending);
+      const response = await submitForm(data);
+
+      if (response.status === 200) {
+        setContactFormState(FormState.success);
+      } else {
+        setContactFormState(FormState.error);
+      }
+    } catch (error) {
+      console.error(error);
+      setContactFormState(FormState.error);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (contactFormState === FormState.success) {
+      setMessageSent(true);
+    }
+  }, [contactFormState]);
+
+  const handleModalClose = useCallback(() => {
+    setMessageSent(false);
+  }, [setMessageSent]);
+
+  const isFormDisabled = useMemo(() => {
+    return (
+      contactFormState === FormState.pending ||
+      contactFormState === FormState.success
+    );
+  }, [contactFormState]);
+
+  return (
+    <div>
+      <form
```

Comment:
add `autocomplete="off"` to prevent input autocomplete

Reply:
done